pkg/config/creds.go

```diff
@@ -61,5 +78,28 @@ func getCredentials(decryptor secrets.Decryptor) (Creds, error) {
 		appKey = val
 	}
 
-	return Creds{APIKey: apiKey, AppKey: appKey}, nil
+	creds := Creds{APIKey: apiKey, AppKey: appKey}
+	cm.cacheCreds(creds)
+	return creds, nil
+}
+
+func (cm *CredentialManager) cacheCreds(creds Creds) {
+	cm.credsMutex.Lock()
+	defer cm.credsMutex.Unlock()
+	cm.creds = creds
+}
+
+func (cm *CredentialManager) getCredsFromCache() (Creds, bool) {
+	cm.credsMutex.Lock()
+	defer cm.credsMutex.Unlock()
+	if cm.creds.APIKey != "" && cm.creds.AppKey != "" {
+		return cm.creds, true
+	}
+	return Creds{}, false
+}
+
+func (cm *CredentialManager) resetCredsCache() {
+	cm.credsMutex.Lock()
+	defer cm.credsMutex.Unlock()
+	cm.creds = Creds{}
 }
```

Comment:
can be removed, because you can avoid using it in the unittest

Reply:
yes good point, done 👍